pkg/landscaper/blueprints/resolve.go

```diff
@@ -27,7 +27,7 @@ import (
 
 // Resolve returns a blueprint from a given reference.
 // If no fs is given, a temporary filesystem will be created.
-func Resolve(ctx context.Context, resolver ctf.ComponentResolver, cdRef *lsv1alpha1.ComponentDescriptorReference, bpDef lsv1alpha1.BlueprintDefinition) (*Blueprint, error) {
+func Resolve(ctx context.Context, resolver ctf.ComponentResolver, cdRef *lsv1alpha1.ComponentDescriptorReference, bpDef lsv1alpha1.BlueprintDefinition, repositoryContext *cdv2.UnstructuredTypedObject) (*Blueprint, error) {
```

Comment:
With this additional parameter, the coding becomes even more complicated especially if you only want to see on its interface. Why not providing just the cdRef with the right repo context. Then the calling method is responsible for providing the right data and this is not mixed into this already very complicated method.